.github/workflows/node-ci-pr.yml

```diff
@@ -165,3 +165,81 @@ jobs:
           pattern: "**/dist/es/*.js"
           # Always ignore SourceMaps and node_modules:
           exclude: "{**/*.map,**/node_modules/**}"
+
+  publish_snapshot:
+    name: Publish npm snapshot
+    # We don't publish snapshots on draft PRs or
+    # on the main Changeset "Version Packages" PR
+    if: |
+      github.event.pull_request.draft == false
+      && github.ref_name != 'changeset-release/main'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node-version: [16.x]
+    steps:
+      # We need to checkout all history, so that the changeset tool can diff it
+      - name: Checkout current commit
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+      - name: Ensure main branch is avaialble
+        run: |
+          REF=$(git rev-parse HEAD)
+          git checkout main
+          git checkout $REF
+      - name: Use Node.js ${{ matrix.node-version }} & Install & cache node_modules
+        uses: Khan/actions@shared-node-cache-v0
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Publish snapshot release to npm
+        id: publish-snapshot
+        run: ./utils/publish/publish-snapshot.sh # All config is via Github env vars
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Calculate short SHA for this commit
+        id: short-sha
+        run: echo "short_sha=$(echo ${GITHUB_SHA} | cut -c1-8)" >> $GITHUB_OUTPUT
+
+      # Note: these two actions are locked to the latest version that were
+      # published when @jeremy (Jeremy Wiebe) created the original job in Perseus.
+      - name: Find existing comment
+        uses: peter-evans/find-comment@034abe94d3191f9c89d870519735beae326f2bdb
+        id: find-comment
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-author: "github-actions[bot]"
+          body-includes: "npm Snapshot:"
+
+      - name: Create or update npm snapshot comment - Success
+        if: steps.publish-snapshot.outputs.npm_snapshot_tag != ''
+        uses: peter-evans/create-or-update-comment@67dcc547d311b736a8e6c5c236542148a47adc3d
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-id: ${{ steps.find-comment.outputs.comment-id }}
+          edit-mode: replace
+          body: |
+            ## npm Snapshot: Published
+            🎉 Good news!! We've packaged up the latest commit from this PR (${{
+            steps.short-sha.outputs.short_sha }}) and published all packages with changesets to npm.
+            You can install it using the tag `${{ steps.publish-snapshot.outputs.npm_snapshot_tag }}`.
+
+            Example:
+            ```sh
+            yarn add @khanacademy/wonder-blocks-switch@${{
+            steps.publish-snapshot.outputs.npm_snapshot_tag }} # Replace `switch` with your component name
+            ```
```

Comment:
**suggestion:** What do you think about adding a comment here including the `deploy_wonder_blocks.js --tag ...` script to explain how this can be used in webapp?

Reply:
makes sense to me! i'll add it 👍